[BUG] 'set' object has no attribute 'extend' when consuming documents

### Description

Document consumption does not work on 1.17.4-dev through docker compose with the image pulled from ghcr.io/paperless-ngx/paperless-ngx:dev

The problem seems to be here https://github.com/paperless-ngx/paperless-ngx/blob/dev/src/documents/data_models.py

### Steps to reproduce

1. put document in the consume folder
2. wait for error to popup in file tasks.

### Webserver logs

```bash
[2023-10-24 17:20:44,008] [DEBUG] [paperless.management.consumer] Waiting for file /usr/src/paperless/consume/test.doc to remain unmodified

[2023-10-24 17:20:49,009] [INFO] [paperless.management.consumer] Adding /usr/src/paperless/consume/test.doc to the task queue.

[2023-10-24 17:53:50,975] [INFO] [celery.worker.strategy] Task documents.tasks.consume_file[e8b4cb0a-7efb-425d-b14a-694ba95772a4] received
[2023-10-24 17:53:52,410] [ERROR] [celery.app.trace] Task documents.tasks.consume_file[e8b4cb0a-7efb-425d-b14a-694ba95772a4] raised unexpected: AttributeError("'set' object has no attribute 'extend'")
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 477, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 760, in __protected_call__
    return self.run(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/tasks.py", line 160, in consume_file
    overrides.update(template_overrides)
  File "/usr/src/paperless/src/documents/data_models.py", line 56, in update
    self.tag_ids.extend(other.tag_ids)
    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'set' object has no attribute 'extend'
```


### Browser logs

_No response_

### Paperless-ngx version

1.17.4-dev

### Host OS

Synology NAS

### Installation method

Docker - official image

### Browser

_No response_

### Configuration changes

_No response_

### Other

_No response_

--- a/src/documents/management/commands/document_consumer.py
+++ b/src/documents/management/commands/document_consumer.py
@@ -32,7 +32,7 @@ except ImportError:  # pragma: nocover
 logger = logging.getLogger("paperless.management.consumer")
 
 
-def _tags_from_path(filepath) -> set[Tag]:
+def _tags_from_path(filepath) -> list[int]:
     """
     Walk up the directory tree from filepath to CONSUMPTION_DIR
     and get or create Tag IDs for every directory.
@@ -47,7 +47,7 @@ def _tags_from_path(filepath) -> set[Tag]:
             Tag.objects.get_or_create(name__iexact=part, defaults={"name": part})[0].pk,
         )
 
-    return tag_ids
+    return list(tag_ids)
 
 
 def _is_ignored(filepath: str) -> bool:

